Doppler.Currency/Services/CurrencyService.cs

```diff
@@ -26,10 +26,10 @@ public class CurrencyService : ICurrencyService
             try
             {
                 _logger.LogInformation("Service Getting currency code handler.");
-                _currencyHandlers.TryGetValue(currencyCode, out var handler);
 
-                if (handler != null)
-                    return await handler.Handle(date);
+                var currencyHandler = _currencyFactory.CreateHandler(currencyCode);
+                if (currencyHandler != null)
```

Comment:
If I am not wrong you should never have a null handler.

Reply:
OK, I will check it and if it is innecesary I will remove the "if"

Reply:
updated